src/gui/menu_species.cpp

```diff
@@ -341,6 +346,82 @@ void DissolveWindow::on_SpeciesSetChargesInSelectionAction_triggered(bool checke
     fullUpdate();
 }
 
+void DissolveWindow::on_SpeciesCopyChargesFromAtomTypesAction_triggered(bool checked)
+{
+    // Get the current Species (if a SpeciesTab is selected)
+    auto species = ui_.MainTabs->currentSpecies();
+    if (!species)
+        return;
+
+    if (QMessageBox::warning(this, "Copy Charges from Atom Types",
+                             "This will replace the species atom charges "
+                             "with those of the assigned atom types.\n\n"
+                             "This cannot be undone! Proceed?",
+                             QMessageBox::StandardButton::Yes | QMessageBox::StandardButton::No,
+                             QMessageBox::StandardButton::No) == QMessageBox::StandardButton::Yes)
+    {
+        const auto atomTypes = dissolve().coreData().atomTypes();
+        for (auto &atom : species->atoms())
+        {
+            for (auto &atomType : atomTypes)
+            {
+                if (atomType == atom.atomType())
+                {
+                    atom.setCharge(atomType->charge());
+                    break;
+                }
+            }
+        }
```

Comment:
Just spotted this one.  Here's another double loop that can be reduced right back to a single one since there's no need to look up the atom type, we can just use it (*if* the atom has one set):
```suggestion
        for (auto &atom : species->atoms())
            if (atom.atomType())
                atom.setCharge(atomType->charge());
```